packages/messaging/lib/messages/DlcOffer.ts

```diff
@@ -133,10 +139,93 @@ export class DlcOfferV0 extends DlcOffer implements IDlcMessage {
     };
   }
 
+  /**
+   * Validates correctness of all fields in DlcOffer
+   * https://github.com/discreetlogcontracts/dlcspecs/blob/master/Protocol.md#the-offer_dlc-message
+   * @throws Will throw an error if validation fails
+   */
   public validate(): void {
+    // 1. Type is set automatically in class
+    // 2. contract_flags field is ignored
+    // 3. chain_hash must be validated as input by end user
+    // 4. payout_spk and change_spk must be standard script pubkeys
+
+    try {
+      address.fromOutputScript(this.payoutSPK);
+    } catch (e) {
+      throw new Error('DlcOffer payoutSPK is invalid');
+    }
+
+    try {
+      address.fromOutputScript(this.changeSPK);
+    } catch (e) {
+      throw new Error('DlcOffer changeSPK is invalid');
+    }
+
+    // 5. funding_pubkey must be a valid secp256k1 pubkey in compressed format
+    // https://github.com/bitcoin/bips/blob/master/bip-0137.mediawiki#background-on-ecdsa-signatures
+
+    if (secp256k1.publicKeyVerify(Buffer.from(this.fundingPubKey))) {
+      if (this.fundingPubKey[0] != 0x02 && this.fundingPubKey[0] != 0x03) {
+        throw new Error('fundingPubKey must be in compressed format');
+      }
+    } else {
+      throw new Error('fundingPubKey is not a valid secp256k1 key');
+    }
+
+    // 6. offer_collateral_satoshis must be greater than or equal to 1000
+
+    if (this.offerCollateralSatoshis < 1000) {
+      throw new Error(
+        'offer_collateral_satoshis must be greater than or equal to 1000',
+      );
+    }
+
+    // 7. cet_locktime and refund_locktime must either both be unix timestamps, or both be block heights.
+    // https://en.bitcoin.it/wiki/NLockTime
+    // https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki#detailed-specification
+    // https://github.com/bitcoin/bitcoin/blob/master/src/script/script.h#L39
+
+    if (
+      !(
+        (this.cetLocktime < LOCKTIME_THRESHOLD &&
+          this.refundLocktime < LOCKTIME_THRESHOLD) ||
+        (this.cetLocktime >= LOCKTIME_THRESHOLD &&
```

Comment:
Nevermind, I see you're just making sure locktime is same unit (timestamp or blocktime) here.